### DISCUSSION TITLE
Why Trainer resume_from_checkpoint only resume when fit.

### DISCUSSION THREAD

**Discussion by tshu-w** [Lightning-AI/pytorch-lightning#8796] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I want to test again with ckpt. However I found Trainer resume_from_checkpoint only work for fit. Is there any particular reason for this?
https://github.com/PyTorchLightning/pytorch-lightning/blob/f3442db3f0155a32c8bc0ca3bc943d4b0039897f/pytorch_lightning/trainer/trainer.py#L840-L841
```
In [1]: from pytorch_lightning import seed_everything, Trainer

In [2]: from src import MMTSMatcher, WDCDataModule

In [3]: trainer = Trainer(resume_from_checkpoint="tests/mmtsmatcher_wdcdatamodule/shoes_small_false_32_0808-182020/checkpoints/epoch=05-valid_f1=81.22%.ckpt", gpus=[4])
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs

In [4]: trainer
Out[4]: <pytorch_lightning.trainer.trainer.Trainer at 0x7f2345178fa0>

In [5]: model = MMTSMatcher(model_name="bert-base-uncased")

In [6]: data = WDCDataModule(training_size="small", cate="shoes")

In [7]: data
Out[7]: <src.datamodules.wdcdatamodule.WDCDataModule at 0x7f22ccd258e0>

In [10]: trainer.test(model, data)
```

**Comment by tchaton** [comment, marked as answer]
Dear @tshu-w,

The resume_start will reload all states which might not be required for testing. We are currently refactoring this part of the code.

Best,
T.C

**Reply by MartaTintore** [reply]
Is there any update on this?

**Reply by rohitgr7** [reply]
@MartaTintore for testing you can pass the ckpt_path inside `trainer.test()`. Other states like optimizer state, scheduler state are not required for testing so `resume_from_checkpoint` might not be the right flag to use while testing.
